### PR TITLE
Adds Restful Client for Scheduler calls

### LINKED ISSUE #2341: Create SchedulerOperations to handle Scheduler to SCDF
The TaskOperations class needs to support scheduler restful client api calls.

This is required for Scheduler AT calls.

### PR DESCRIPTION
**Pull request by cppwfs**
Also DataFlowTemplate tests.

resolves #2341

### CODE REVIEW

File: spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/SchedulerTemplate.java [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+/**
+ * Implementation for
+ * {@link org.springframework.cloud.dataflow.rest.client.SchedulerOperations}.
+ *
+ * @author Glenn Renfro
+ */
+public class SchedulerTemplate implements SchedulerOperations {
+
+	/* default */ static final String SCHEDULES_RELATION = "tasks/schedules";
```
Comment (markpollack) [inline spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/SchedulerTemplate.java]: seems we make these public or private in other similar template implementations...

File: spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/SchedulerTests.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.RestTemplate;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+/**
+ * @author Glenn Renfro
+ */
+public class SchedulerTests {
```
Comment (markpollack) [inline spring-cloud-dataflow-rest-client/src/test/java/org/springframework/cloud/dataflow/rest/client/SchedulerTests.java]: SchedulerTemplateTests instead?

**Comment by cppwfs** [comment]
Rebased

**Comment by sabbyanandan** [comment]
This is already merged in https://github.com/spring-cloud/spring-cloud-dataflow/commit/1cba8b787a896c18c45915c677307f0ac1b101fa.

**Pull request closed without merging**